cache.go

```diff
@@ -255,6 +275,10 @@ func (c *Cache) SetWithTTL(key, value interface{}, cost int64, ttl time.Duration
 		return false
 	}
 
+	if c.keySaver != nil {
+		c.keySaver.AddKey(fmt.Sprintf("%v", key))
```

Comment:
It could be possible that the item is rejected by the cache if after this point if
1. We're under heavy load or
2. The cost of item exceeds the max cost of the cache.

This isn't the right place to Add the key to the keySaver.
@Darkweak I think the keysaver stuff doesn't need to be in ristretto. You can keep the keysaver in your application code.
Here's how you would do it
```
cache.Evict = func(item *ristretto.Item) { keySaver.Del(item.Key()) }
if cache.Set(...,...) { keySaver.Add(....) }
```

Reply:
Thank you for your response, IMO we can close this PR and i'll implement your solution and support it directly in my application. You're right, ristretto shouldn't support that and could decrease performances too.

Thanks for the work you're doing on ristretto 👍

Reply:
Why item only have keyHash. The key was lost. Annd when `cache.Set` we can't get keyHash from cache. So how can I got key by `item.Key()`?